snapcraft/plugins/plainbox_provider.py

```diff
@@ -43,6 +44,13 @@ def __init__(self, name, options, project):
 
     def build(self):
         super().build()
+        env = os.environ.copy()
+        provider_stage_dir = os.path.join(self.project.stage_dir, "providers")
+        if os.path.exists(provider_stage_dir):
+            providerdirs = [os.path.join(provider_stage_dir, provider)
+                            for provider in os.listdir(provider_stage_dir)]
+            env['PROVIDERPATH'] = ':'.join(providerdirs)
+        self.run(["python3", "manage.py", "validate"], env=env)
```

Comment:
There is a unit test that checks these calls. You can add validate in there.
And you can also add a couple of unit tests for the env, one with provider_stage_dir, and one without it.